validator/README.md

```diff
@@ -0,0 +1,78 @@
+# Validator
+
+## About
+
+We use [go-playground/validator.v9](https://github.com/go-playground/validator/tree/v9)
+which provides flexible and rich methods to validate structs and values for validation.
+
+## Design
+
+Validator makes use of `Operators` in `definition.Parameter` struct.
+
+`type Operator func(ctx context.Context, object interface{}) (interface{}, error)`
+
+Each validator is a Operator. Nirvana provided 3 preset validators which using validator.v9 inside,
+so you can use [validators and tags provided by go-playground/validator.v9](https://godoc.org/gopkg.in/go-playground/validator.v9#hdr-Baked_In_Validators_and_Tags) directly.
+
+`validator.Struct` for struct value validation.
+
+`validator.Var` and `validator.VarWithValue` for single variable validation.
+
+## Custom Validator
+
+More complicated validation can be achieved by writing your own validator like this:
+
+```
+type Booking struct {
+	UserName string    `json:"name" validate:"required,printascii"`
+	CheckIn  time.Time `json:"checkIn" validate:"required"`
+	CheckOut time.Time `json:"checkOut" validate:"required,gtfield=CheckIn"`
+}
+
+func validateBooking(ctx context.Context, obj interface{}) (interface{}, error) {
+	// Note we can still use validator pkg to check before our custom validation code.
+	obj, err := validator.Struct()(ctx, obj)
```

Comment:
What is `validator`? A package, or a variable?

Reply:
@walktall The README doesn't provide necessary background.  I'm gonna merge it first.  We'll do documentation during end of Dec.